packages/thrift-server-core/src/main/protocols/BinaryProtocol.ts

```diff
@@ -283,9 +288,15 @@ export class BinaryProtocol extends TProtocol {
         return this.transport.readI32()
     }
 
-    public readI64(): Int64 {
+    public readI64(type?: 'int64'): Int64
+    public readI64(type: 'bigint'): bigint
+    public readI64(type?: I64Type): bigint | Int64 {
```

Comment:
apart from the naming, whats the point of returning a `bigint`?

Reply:
it seems weird to me to call `readI64` and getting a bigint in return.
i'm also not sure what the usecase is that you would want a bigint, when it'll only even have 64bits of precision

Reply:
`readI64` here refers purely to the Thrift type we're reading from the Buffer, not the the return type. The goal here is to allow users the option to use a JavaScript primitive value that will satisfy the needs of storing an `i64` without having to rely on an external library. It is difficult to do anything useful with an `Int64` whereas a `BigInt` just works with all the usual JS operators.